src/rust/src/backend/ec.rs

```diff
@@ -187,6 +187,8 @@ fn derive_private_key(
     let ec = openssl::ec::EcKey::from_private_components(&curve, &private_value, &point)
         .map_err(|_| pyo3::exceptions::PyValueError::new_err("Invalid EC key"))?;
     check_key_infinity(&ec)?;
+    ec.check_key()
```

Comment:
Does this also do the infinity check?

Reply:
It does, yes.

Reply:
So then should we remove `check_key_infinity` on line 189?

Reply:
done